hilti/toolchain/include/ast/expressions/builtin-function.h

```diff
@@ -0,0 +1,83 @@
+// Copyright (c) 2020-2021 by the Zeek Project. See LICENSE for details.
+
+#include <string>
+#include <utility>
+#include <vector>
+
+#include <hilti/ast/declarations/parameter.h>
+#include <hilti/ast/expression.h>
+#include <hilti/ast/node.h>
+#include <hilti/ast/type.h>
+
+namespace hilti {
+namespace expression {
+
+/** AST node representing a builtin function. */
+class BuiltinFunction : public NodeBase, public trait::isExpression {
+public:
+    BuiltinFunction(std::string name, std::string cxxname, hilti::Type type,
+                    std::vector<declaration::Parameter> parameters, std::vector<Expression> arguments, Meta m = Meta())
+        : NodeBase(nodes(arguments), std::move(m)),
+          _name(std::move(name)),
+          _cxxname(std::move(cxxname)),
+          _type(std::move(type)),
```

Comment:
Type should normally be child node as well so that it participates in AST traversal (and hence type resolution). Any particular reason for not making it one?